File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.py
from functools import cmp_to_key

def solution(numbers):
    numbers = list(map(str, numbers))
    
    def compare(x, y):
        if x + y > y + x:
            return -1  
        elif x + y < y + x:
            return 1 
        else:
            return 0 
    
    numbers.sort(key=cmp_to_key(compare))
    
    result = ''.join(numbers)
    
    return result if result[0] != '0' else '0'

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.cpp
#include <bits/stdc++.h>
using namespace std;
bool cmp(string a, string b){
    return a + b > b + a;
}
string solution(vector<int> numbers) {
    string answer = "";
    vector<string> s;
    
    for(int i = 0; i < numbers.size(); i++) s.push_back(to_string(numbers[i]));
    
    sort(s.begin(), s.end(), cmp);
    if(s[0].compare("0") == 0) return "0";
    for(int i = 0 ;i < s.size(); i++) answer += s[i];
    return answer;
}